src/Cli/Microsoft.TemplateEngine.Cli/Commands/NewCommand.cs

```diff
@@ -93,6 +93,16 @@ internal partial class NewCommand : BaseCommand<NewCommandArgs>, ICustomHelp
             Hidden = true
         };
 
+        // This is only used to let the help generation know there is a name argument.
+        // It doesn't contribute to parsing the arguments because the arity is set to 0.
+        // InstantiateCommand reparses the arguments in the context of TemplateCommand to properly parse the name.
+        internal static CliArgument<string> NameArgument { get; } = new CliArgument<string>("name")
+        {
+            Description = SymbolStrings.TemplateCommand_Option_Name,
+            Arity = new ArgumentArity(0, 0),
+            Hidden = true
+        };
```

Comment:
Workaround to make help generate properly.

Reply:
Did some investigation with this. It looks like both `dotnet new --help` and `dotnet new create --help` both call this `CustomUsageSection` method. Without my code above, `new` is missing the `name` argument information, but `create` is not. This is because `InstantiateCommand` (aka `create`) has the *real* name argument, but we don't want a *real* name argument for `NewCommand`. We only want to show it in the help for `new`.

The `CustomUsageSection` method is for a custom *Usage* section but doesn't cover the *Arguments* section, which is currently not custom. That's why I opted to create this 'fake' name argument since it would "just work" for both sections. If I was to touch this help output, I'd likely refactor more than just working around this situation. But I didn't want to do all that for this PR.